search/api.py

```diff
@@ -0,0 +1,261 @@
+            ))
+
+    _CONN_VERIFIED = True
+    return _CONN
+
+
+def _index_users_chunk(users):
+    """
+    Add/update a small number of user records in Elasticsearch
+
+    Args:
+        users (list of User):
+            List of users
+
+    Returns:
+        int: Number of items inserted into Elasticsearch
+    """
+
+    conn = get_conn()
+    insert_count, errors = bulk(
+        conn,
+        (serialize_user(user) for user in users),
+        index=settings.ELASTICSEARCH_INDEX,
+        doc_type=USER_DOC_TYPE,
+    )
+
+    if len(errors) > 0:
+        raise ReindexException("Error during bulk insert: {errors}".format(
+            errors=errors
+        ))
+    refresh_index()
+
+    return insert_count
+
+
+def index_users(users, chunk_size=100):
+    """
+    Add/update profile records in Elasticsearch.
+
+    Args:
+        users (iterable of User):
+            Iterable of users
+        chunk_size (int):
+            How many users to index at once.
+
+    Returns:
+        int: Number of indexed items
+    """
+    # Use an iterator so we can keep track of what's been indexed already
+    users = iter(users)
+
+    count = 0
+    chunk = list(islice(users, chunk_size))
+    while len(chunk) > 0:
+        count += _index_users_chunk(chunk)
+        chunk = list(islice(users, chunk_size))
+
+    refresh_index()
+
+    return count
+
+
+def remove_user(user):
+    """
+    Remove a user from Elasticsearch.
+    """
+    conn = get_conn()
+    try:
+        conn.delete(index=settings.ELASTICSEARCH_INDEX, doc_type=USER_DOC_TYPE, id=user.id)
+    except NotFoundError:
+        # Item is already gone
+        pass
+
+
+def serialize_user(user):
+    """
+    Serializes user for use with Elasticsearch.
+
+    Args:
+        user (User): A user to serialize
+    Returns:
+        dict: The data to be sent to Elasticsearch
+    """
+    serialized = {
+        'id': user.id,
+        '_id': user.id,
+    }
+    try:
+        serialized['profile'] = ProfileSerializer().to_representation(user.profile)
+    except Profile.DoesNotExist:
+        # Just in case
+        pass
+    return serialized
+
+
+def refresh_index():
+    """
+    Refresh the Elasticsearch index
+    """
+    get_conn().indices.refresh(index=settings.ELASTICSEARCH_INDEX)
+
+
+def create_user_mapping():
+    """
+    Create a mapping for profiles. If one already exists, delete it first.
+    """
+    conn = get_conn(verify=False)
+
+    index_name = settings.ELASTICSEARCH_INDEX
+    if conn.indices.exists_type(index=index_name, doc_type=USER_DOC_TYPE):
+        conn.indices.delete_mapping(index=index_name, doc_type=USER_DOC_TYPE)
+
+    # A string which must be indexed exactly as is and not stemmed for full text search (the default)
+    not_analyzed_string_type = {
+        'type': 'string',
+        'index': 'not_analyzed',
+    }
+    bool_type = {'type': 'boolean'}
+    date_type = {'type': 'date', 'format': 'date'}
+    long_type = {'type': 'long'}
```

Comment:
if we're setting `USER_DOC_TYPE` as a module-level variable, i think it makes sense to do the same for the above 4 values